### PR TITLE
Matched some constructors in d_tree.cpp and made progress on dTree_room_c::newData(), dTree_room_c::deleteData()

### PR DESCRIPTION
**Pull request by link5669**
Matched:

- dTree_room_c::newData()
- dTree_room_c::dTree_room_c()
- dTree_data_c::dTree_data_c()

Progress:

- dTree_packet_c::dTree_packet_c()
- void dTree_room_c::deleteData()

### CODE REVIEW

File: src/d/actor/d_a_pirate_flag.cpp [hunk 1]
```diff
@@ -52,8 +52,8 @@ void daPirate_Flag_Execute(pirate_flag_class*) {
 }
 
 /* 00001A38-00001A40       .text daPirate_Flag_IsDelete__FP17pirate_flag_class */
-void daPirate_Flag_IsDelete(pirate_flag_class*) {
-    /* Nonmatching */
+int daPirate_Flag_IsDelete(pirate_flag_class*) {
+    return 1;
```
Comment (magcius) [inline src/d/actor/d_a_pirate_flag.cpp:56]: This should be BOOL and it should return TRUE

File: src/d/actor/d_a_pirate_flag.cpp [hunk 2]
```diff
@@ -23,7 +23,7 @@ void daPirate_Flag_packet_c::setNrmVtx(cXyz*, int, int) {
 
 /* 0000083C-00000E44       .text draw__22daPirate_Flag_packet_cFv */
 BOOL daPirate_Flag_packet_c::draw() {
-    /* Nonmatching */
+    return true;
```
Comment (magcius) [inline src/d/actor/d_a_pirate_flag.cpp:26]: fyi, we're not perfect at this, but the usual convention is that BOOL goes with TRUE/FALSE, while bool goes with true/false

File: src/d/actor/d_a_pirate_flag.cpp [hunk 3]
```diff
@@ -23,7 +23,8 @@ void daPirate_Flag_packet_c::setNrmVtx(cXyz*, int, int) {
 
 /* 0000083C-00000E44       .text draw__22daPirate_Flag_packet_cFv */
 BOOL daPirate_Flag_packet_c::draw() {
-    /* Nonmatching */
```
Comment (LagoLunatic) [inline src/d/actor/d_a_pirate_flag.cpp:26]: it's best to leave the Nonmatching comment in until the function is fully matched so it's clear at a glance it needs more work

File: src/d/actor/d_a_tag_mk.cpp [hunk 4]
```diff
@@ -7,18 +7,18 @@
...
-    /* Nonmatching */
+unsigned int daTag_Mk_c::getSwbit2() {
+    return this->mBase.mParameters & 0xff;
 }
 
 /* 00000090-0000009C       .text getType__10daTag_Mk_cFv */
-void daTag_Mk_c::getType() {
-    /* Nonmatching */
+unsigned int daTag_Mk_c::getType() {
+    return this->mBase.mParameters & 0xff;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tag_mk.cpp:21]: are these three parameter functions supposed to all be exactly the same?

File: src/d/actor/d_a_pirate_flag.cpp [hunk 5]
```diff
@@ -23,7 +23,8 @@ void daPirate_Flag_packet_c::setNrmVtx(cXyz*, int, int) {
...
+    //temp return to allow to build
+    return TRUE;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_pirate_flag.cpp:27]: you shouldn't need to add returns just for it to build, it should only give a warning if a non-void function is missing a return value. if you're getting an error it's probably caused by something else.

File: src/d/actor/d_a_tag_mk.cpp [hunk 6]
```diff
@@ -122,8 +122,8 @@ void daTag_Mk_c::actionArrival() {
 }
 
 /* 00000A48-00000A50       .text actionWait__10daTag_Mk_cFv */
-void daTag_Mk_c::actionWait() {
-    /* Nonmatching */
+int daTag_Mk_c::actionWait() {
+    return 1;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tag_mk.cpp:126]: all these int/1 functions are probably BOOL/TRUE

**Comment by link5669** [comment]
Sorry, the "cname" commit was intended for another repo. That commit was making progress on d_a_mmusic.cpp

File: src/d/actor/d_a_mmusic.cpp [hunk 7]
```diff
@@ -4,21 +4,40 @@
...
+
+    if ((((iVar1 == 0) &&
+          (iVar1 = g_dComIfG_gameInfo.save.mSavedata.mEvent.isEventBit(0x2910), iVar1 == 0)) &&
+         (iVar1 = g_dComIfG_gameInfo.save.mSavedata.mEvent.isEventBit(0x2e02), iVar1 == 0)) &&
+        ((iVar1 = g_dComIfG_gameInfo.save.mSavedata.mEvent.isEventBit(0x1610),
+          iVar1 == 0 && (bVar2 = dComIfGs_checkGetItem(LV3_SWORD), bVar2))))
+    {
+        return TRUE;
+    }
+    return FALSE;
```
Comment (magcius) [inline src/d/actor/d_a_mmusic.cpp:40]: I would write this as:

```cpp
if (!dComIfGs_isStageBossEnemy(7) && !dComIfGs_isEventBit(0x2910)  && !dComIfGs_isEventBit(0x2e02)  && !dComIfGs_isEventBit(0x1610) && dComIfGs_checkGetItem(LV3_SWORD))
    return TRUE;
else
    return FALSE;
```

File: src/d/d_timer.cpp [hunk 8]
```diff
@@ -32,8 +33,10 @@ void dTimer_c::RestTimeCheck(int) {
 }
 
 /* 8023C110-8023C124       .text deleteCheck__8dTimer_cFv */
-void dTimer_c::deleteCheck() {
-    /* Nonmatching */
+unsigned int dTimer_c::deleteCheck() {
+    unsigned int uVar1;
+    // uVar1 = std::countl_zero(5 - (unsigned int)(char)this->field_0x164);
+    return uVar1 >> 5 & 0xff;
```
Comment (magcius) [inline src/d/d_timer.cpp:39]: This is probably `return field_0x164 == 0;`

File: src/d/d_timer.cpp [hunk 9]
```diff
@@ -172,27 +180,27 @@ void dDlst_TimerScrnDraw_c::draw() {
 }
 
 /* 8023D9A0-8023D9C0       .text dTimer_Draw__FP8dTimer_c */
-void dTimer_Draw(dTimer_c*) {
-    /* Nonmatching */
+void dTimer_Draw(dTimer_c* timer) {
+    timer->_draw();
```
Comment (magcius) [inline src/d/d_timer.cpp:184]: Should be
```cpp
bool dTimer_Draw(dTimer_c* timer) {
    return timer->_draw();
}
```

and also for all the others below.

File: src/d/d_timer.cpp [hunk 10]
```diff
@@ -208,6 +216,15 @@ void dTimer_createStockTimer() {
...
+    if (this != (dDlst_TimerScrnDraw_c*)0x0) {
+        // (this->parent).vtbl = &__vt;
+        if (this != (dDlst_TimerScrnDraw_c*)0x0) {
+            // (this->parent).vtbl = (undefined*)&dDlst_base_c::__vt;
+        }
+        if (0 < in_r4) {
+            // delete((JKRHeap*)this);
+        }
+    }
+    return;
```
Comment (magcius) [inline src/d/d_timer.cpp:229]: I suspect most of this is autogenerated. Try just an empty function here.

File: src/d/d_tree.cpp [hunk 11]
```diff
@@ -32,33 +33,52 @@ void dTree_data_c::animation(int) {
...
+    this->field_0x6736 = 0;
+    pdVar1 = this->mData;
+    iVar4 = 0x40;
+    return;
 }
 
 /* 800795E8-800795F4       .text __ct__12dTree_room_cFv */
 dTree_room_c::dTree_room_c() {
-    /* Nonmatching */
+    this->mpHead = (dTree_data_c*)0x0;
```
Comment (magcius) [inline src/d/d_tree.cpp:69]: `mpHead = NULL;`

Get rid of the extra this-> and return;s please

File: src/d/d_tree.cpp [hunk 12]
```diff
@@ -32,33 +33,52 @@ void dTree_data_c::animation(int) {
...
+    dTree_anm_c* pdVar2;
+    short sVar3;
+    int iVar4;
+
+    this->field_0x0010 = 0;
+    this->field_0x6734 = 0;
+    this->field_0x6736 = 0;
+    pdVar1 = this->mData;
+    iVar4 = 0x40;
+    return;
```
Comment (magcius) [inline src/d/d_tree.cpp:64]: Please leave the Nonmatching in if it doesn't match. This doesn't even look close. The loop below is probably something like:

```cpp
for (s32 i = 0; i < (s32)ARRAY_SIZE(mData); i++)
    mData[i].field_0x000 = 0;
```

File: src/d/d_tree.cpp [hunk 13]
```diff
@@ -32,33 +33,52 @@ void dTree_data_c::animation(int) {
...
 dTree_room_c::dTree_room_c() {
-    /* Nonmatching */
+    this->mpHead = (dTree_data_c*)0x0;
+    return;
 }
 
 /* 800795F4-80079600       .text __ct__11dTree_anm_cFv */
 dTree_anm_c::dTree_anm_c() {
-    /* Nonmatching */
+    return;
```
Comment (magcius) [inline src/d/d_tree.cpp:75]: Remove empty return

File: include/d/d_tree.h [hunk 14]
```diff
@@ -4,6 +4,12 @@
 #include "SSystem/SComponent/c_xyz.h"
 #include "dolphin/mtx/mtx.h"
 #include "JSystem/J3DGraphBase/J3DPacket.h"
+#include "JSystem/JKernel/JKRHeap.h"
+#include "JSystem/JUtility/JUTAssert.h"
+#include "d/actor/d_a_obj_pirateship.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_procname.h"
+#include "d/d_s_play.h"
```
Comment (magcius) [inline include/d/d_tree.h:12]: These should go in the .cpp, not the .h

File: include/d/actor/d_a_mmusic.h [hunk 15]
```diff
@@ -4,23 +4,29 @@
...
-        void delete_se();
-        BOOL _execute();
-        BOOL _draw();
-    
-    public:
-        /* Place member variables here */
-    };
+class Act_c : public fopAc_ac_c {
+public:
+    bool solidHeapCB(fopAc_ac_c*);
```
Comment (magcius) [inline include/d/actor/d_a_mmusic.h:9]: solidHeapCB is probably a static, and takes the actor as the first parameter

File: src/d/actor/d_a_mmusic.cpp [hunk 16]
```diff
@@ -4,21 +4,40 @@
...
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_mtx.h"
 
 /* 00000078-0000009C       .text solidHeapCB__Q28daMmusic5Act_cFP10fopAc_ac_c */
-void daMmusic::Act_c::solidHeapCB(fopAc_ac_c*) {
+bool daMmusic::Act_c::solidHeapCB(fopAc_ac_c* param) {
     /* Nonmatching */
+    bool uVar1;
+    uVar1 = create_heap();
+    return uVar1;
```
Comment (magcius) [inline src/d/actor/d_a_mmusic.cpp:21]: See other files for this, but it's probably something like:

```cpp
BOOL daMmusic::Act_c::solidHeapCB(fopAc_ac_c* i_this) {
    return ((Act_c*)i_this)->create_heap();
}
```

File: src/d/actor/d_a_mmusic.cpp [hunk 17]
```diff
@@ -32,13 +51,22 @@ s32 daMmusic::Act_c::_create() {
...
+        *(unsigned int*)(field_0x2A0) = 0;
+    }
+    delete_se();
+    return true;
 }
 
 /* 000002B4-000002C0       .text init_se__Q28daMmusic5Act_cFv */
 void daMmusic::Act_c::init_se() {
     /* Nonmatching */
+    (this->field_0x2D4) = 0x78;
```
Comment (magcius) [inline src/d/actor/d_a_mmusic.cpp:69]: What's not matching about this? Given how the value is 0x78 / 120 (4 seconds at 30fps), I'm guessing this is a timer.

File: src/d/actor/d_a_mmusic.cpp [hunk 18]
```diff
@@ -4,21 +4,40 @@
...
-void daMmusic::Act_c::create_heap() {
-    /* Nonmatching */
+bool daMmusic::Act_c::create_heap() {
+    return true;
 }
 
 /* 000000A4-0000013C       .text Macore_is_playing__Q28daMmusic5Act_cFv */
-void daMmusic::Act_c::Macore_is_playing() {
+bool daMmusic::Act_c::Macore_is_playing() {
     /* Nonmatching */
```
Comment (magcius) [inline src/d/actor/d_a_mmusic.cpp:31]: What's not matching about this?

File: src/d/actor/d_a_tag_mk.cpp [hunk 19]
```diff
@@ -7,18 +7,20 @@
...
+unsigned int daTag_Mk_c::getSwbit2() {
     /* Nonmatching */
+    return this->mBase.mParameters & 0xff;
 }
 
 /* 00000090-0000009C       .text getType__10daTag_Mk_cFv */
-void daTag_Mk_c::getType() {
-    /* Nonmatching */
+unsigned int daTag_Mk_c::getType() {
+    return this->mBase.mParameters & 0xff;
```
Comment (magcius) [inline src/d/actor/d_a_tag_mk.cpp:23]: Besides what Lago said, `this->mBase.mParameters` is most likely `fopAcM_GetParam(this)`

File: src/d/actor/d_a_mmusic.cpp [hunk 20]
```diff
@@ -32,13 +51,22 @@ s32 daMmusic::Act_c::_create() {
 }
 
 /* 00000268-000002B4       .text _delete__Q28daMmusic5Act_cFv */
-BOOL daMmusic::Act_c::_delete() {
+bool daMmusic::Act_c::_delete() {
     /* Nonmatching */
+    int iVar1;
+    if (this->field_0x2A0 != 0) {
```
Comment (magcius) [inline src/d/actor/d_a_mmusic.cpp:57]: Looking a bit more at the actor, _execute sets this field to a JPABaseEmitter*, so I'm guessing this is:

```cpp
bool daMmusic::Act_c::_delete() {
    if (mEmitter != NULL) {
        mEmitter->becomeInvalidEmitter();
        mEmitter = NULL;
    }
    delete_se();
    return true;
}
```

which is a pattern that shows up elsewhere.

File: src/d/d_timer.cpp [hunk 21]
```diff
@@ -121,14 +126,21 @@ void dDlst_TimerScrnDraw_c::setRupee(short) {
...
-/* 8023CF48-8023CF98       .text setPaneInitialPos__21dDlst_TimerScrnDraw_cFP18fopMsgM_pane_classff */
-void dDlst_TimerScrnDraw_c::setPaneInitialPos(fopMsgM_pane_class*, float, float) {
+/* 8023CF48-8023CF98       .text setPaneInitialPos__21dDlst_TimerScrnDraw_cFP18fopMsgM_pane_classff
+ */
+void dDlst_TimerScrnDraw_c::setPaneInitialPos(fopMsgM_pane_class* param_1, float param_2,
+                                              float param_3) {
     /* Nonmatching */
+    (param_1->mPosCenterOrig).x = param_2 + (0.5 * (param_1->mSizeOrig).x);
+    (param_1->mPosCenterOrig).y = param_3 + (param_1->mSizeOrig).y * 0.5;
+    fopMsgM_paneTrans(param_1, 0.0, 0.0);
```
Comment (magcius) [inline src/d/d_timer.cpp:142]: Heads up, it's probably not going to match unless you include the f suffix on floats -- e.g. `param_2 + (0.5f * param_1->mSizeOrig.x)`. You should include the f suffix on basically every float.

**Pull request closed without merging**